backend/src/nodes/image_iterator_nodes.py

```diff
@@ -100,12 +100,13 @@ def walk_error_handler(exception_instance):
             )
 
         just_image_files: List[str] = []
-        for root, _dirs, files in os.walk(
+        for root, dirs, files in os.walk(
             directory, topdown=True, onerror=walk_error_handler
         ):
             await context.progress.suspend()
 
-            for name in sorted(files):
+            dirs.sort(key=numerical_sort)
```

Comment:
We don't use `dirs`, so why are you sorting it?

Reply:
We don't, os.walk does. Try walking a tree with numbered directories. If you don't sort the directories, they'll come out e.g. 1, 10, 11, 12, 7, 8, 9. With that sort, it will walk them 1, 7, 8, 9, 10, 11, 12.

Reply:
So I looked up the behavior of `os.walk` and you are right. I did not expect that `dirnames` can be modified. I would proclaim that iterators changing their behavior based on what happens to their output is an antipattern (or as some people call it: pythonic).

(Could you please resolve this thread? I can't do that myself...)

Reply:
It's definitely weird. I can't think of another function that does that.

Reply:
In addition, sorting is different in linux and windows.